src/User/ProfilePage.test.jsx

```diff
@@ -16,32 +16,134 @@ jest.mock("react-router-dom", () => {
 configure({ adapter: new Adapter() });
 
 describe("ProfilePage", () => {
-    const mockData = {
-        id: '11',
-        username: "RMorey",
-        firstName: "Ross",
-        lastName: "Morey",
-        email: "Ross.Morey@live.co.uk",
-        occupation: "Fun" 
+  const mockDataWithManager = {
+    id: 11,
+    username: "RMorey",
+    firstName: "Ross",
+    lastName: "Morey",
+    email: "Ross.Morey@live.co.uk",
+    occupation: "Fun",
+    managerId: 1
   };
 
-  it("renders profile information", async () => {
-    expect.assertions(5);
+  const mockDataWithoutManager = {
+    id: 11,
+    username: "RMorey",
+    firstName: "Ross",
+    lastName: "Morey",
+    email: "Ross.Morey@live.co.uk",
+    occupation: "Fun",
+    managerId: null
+  };
+
+  const mockManagerName = "Joe Bloggs";
+
+  const mockNoManagerName = "None Assigned";
+
+  let wrapper;
+  const findElement = identifier => {
+    return wrapper.find(identifier).get(0).props.children;
+  };
+
+  it("renders current user's profile information with manager", async () => {
+    expect.assertions(7);
+
+    const history = createMemoryHistory();
+    history.push("/users/11");
+
+    wrapper = mount(
+      <Router history={history}>
+        <ProfilePage
+          user={mockDataWithManager}
+          isCurrentUser={true}
+          managerName={mockManagerName}
+        />
+      </Router>
+    );
+
+    wrapper.update();
+    expect(wrapper.find(".editLink").get(0).props.to).toBe(
+      `/users/${mockDataWithManager.id}/edit`
+    );
+    expect(findElement(".name")).toBe(
+      `Hello, ${mockDataWithManager.firstName} ${mockDataWithManager.lastName}!`
```

Comment:
You can expect this to be `Hello, Ross Morey` i dont think you really need to read from the mock here like this. But its more a personal opinion on testing
We do it a bit of both ways in SUTD

Reply:
I left it like this cause I figured it'd stop the test breaking in the future if someone changed the mock data.